registry/infrastructure/repositories/organization_repository.py

```diff
@@ -1,529 +1,132 @@
-    pass
+class OrganizationPublisherRepository(BaseRepository):
+
+    def get_org(self, org_uuid):
+        organization = self.session.query(Organization).filter(Organization.uuid == org_uuid).first()
+        if organization is None:
+            raise OrganizationNotFoundException()
+        organization_domain_entity = OrganizationFactory.org_domain_entity_from_repo_model(organization)
+        self.session.commit()
+        return organization_domain_entity
+
+    def get_org_for_user(self, username):
+        organizations = self.session.query(Organization) \
+            .join(OrganizationMember, Organization.uuid == OrganizationMember.org_uuid) \
+            .filter(OrganizationMember.username == username).all()
+        organizations_domain_entity = OrganizationFactory.org_domain_entity_from_repo_model_list(organizations)
+        self.session.commit()
+        return organizations_domain_entity
+
+    def store_ipfs_hash(self, organization, username):
+        organization_db_model = self.session.query(Organization).filter(Organization.uuid == organization.uuid).first()
+        organization_db_model.assets = organization.assets
+        organization_db_model.metadata_ipfs_hash = organization.metadata_ipfs_hash
+        organization_db_model.org_state[0].updated_on = datetime.utcnow()
+        organization_db_model.org_state[0].updated_by = username
+        self.session.commit()
+
+    def persist_publish_org_transaction_hash(self, org_uuid, transaction_hash, wallet_address, username):
+        organization_db_model = self.session.query(Organization).filter(Organization.uuid == org_uuid).first()
+        if organization_db_model is None:
+            raise OrganizationNotFoundException()
+        organization_db_model.org_state[0].trasaction_hash = transaction_hash
+        organization_db_model.org_state[0].wallet_address = wallet_address
+        organization_db_model.org_state[0].state = OrganizationStatus.PUBLISH_IN_PROGRESS.value
+        organization_db_model.org_state[0].updated_on = datetime.utcnow()
+        organization_db_model.org_state[0].updated_by = username
+
+    def store_organization(self, organization, username, state):
+        organization_db_model = self.session.query(Organization).filter(
+            Organization.uuid == organization.uuid).first()
+        if state in [OrganizationStatus.DRAFT.value, OrganizationStatus.APPROVAL_PENDING.value]:
+            if organization_db_model is None:
+                self.add_organization(organization, username, state)
+            else:
+                self.update_organization(organization_db_model, organization, username, state)
+
+    def add_organization(self, organization, username, state):
+        current_time = datetime.utcnow()
+        org_addresses_domain_entity = organization.addresses
+        group_domain_entity = organization.groups
+
+        groups = [
+            Group(name=group.name, id=group.group_id, org_uuid=organization.uuid,
```

Comment:
we cab extract these constructions of db_models in some common place as we might need to reuse it.

Reply:
better to extract it when usage occurs